src/finalisers/iife.ts

```diff
@@ -80,7 +80,7 @@ export default function iife (
 
 	if (isNamespaced) {
 		wrapperIntro =
-			setupNamespace(name, 'this', false, options.globals) + wrapperIntro;
+			setupNamespace(name, `${typeof self !== 'undefined' ?  self : typeof global !== 'undefined' ? global : this}`, false, options.globals) + wrapperIntro;
```

Comment:
This does not work. This is merely a string cast of the previous code while we actually need a string here. Also this should respect the new "compact" option.